estimagic/optimization/nag_optimizers.py

```diff
@@ -0,0 +1,354 @@
+            ``20 * len(x)``.
+        comparison_period_for_insufficient_improvement (int):
+            How many iterations to go back to calculate the improvement.
+            For example 5 would mean that each criterion evaluation is compared to the
+            criterion value from 5 iterations before.
+        quit_when_trust_evaluations_within_noise_level (bool): Flag to quit
+            (or restart) if all $f(y_t)$ are within noise level of
+            $f(x_k)$. Default is ``True`` if ``noisy_criterion`` and
+            ``False`` else.
+        noise_scale_factor_for_quit (float): Factor of the noise level to use in
+            termination criterion.
+        multiplicative_noise_level (float): Multiplicative noise level in the
+            criterion. You can only specify ``multiplicative_noise_level`` or
+            ``additive_noise_level``.
+        additive_noise_level (float): Additive noise level in the
+            criterion. You can only specify ``multiplicative_noise_level`` or
+            ``additive_noise_level``.
+        scale_interpolation_system (bool): whether or not to scale the interpolation
+            linear system to improve conditioning.
+        frobenius_for_interpolation_problem (bool): whether to solve the
+            underdetermined quadratic interpolation problem by minimizing the Frobenius
+            norm of the Hessian, or change in Hessian.
+        use_restarts (bool): whether to do restarts when the lower bound on the trust
+            region radius (:math:`\rho_k`) reaches the stopping criterion
+            (:math:`\rho_{end}`), or (optionally) when all points are within noise
+            level. Default is ``True`` if ``criterion_noisy`` or when
+            ``seek_global_optimum``.
+        max_unsuccessful_restarts (int): maximum number of consecutive unsuccessful
+            restarts allowed (i.e. restarts which did not reduce the objective further)
+        max_unsuccessful_restarts_total (int): number of total unsuccessful restarts
+            allowed. Default is 20 if ``seek_global_optimum`` and else unrestricted.
+        trust_region_rescaling_after_unsuccessful_restart (float): Factor by which to
+            increase the initial trust region radius (:math:`\rho_{beg}`) after
+            unsuccessful restarts. Default is 1.1 if ``seek_global_optimum`` else 1.
+        trust_region_stop_criterion_scaling_after_restart (float): Factor with which
+            the trust region stopping criterion is multiplied with at each restart.
+        use_soft_restarts (bool): Whether to use soft or hard restarts.
+        nr_points_to_move_at_soft_restart (int): Number of interpolation points to
+            move at each soft restart.
+        move_current_point_at_soft_restart (bool): Whether to move the current
+            evaluation point ($x_k$) to the best new point evaluated.
+        reuse_criterion_value_at_hard_restart (bool): whether or not to recycle the
```

Comment:
My guess is it saves one function evaluation to not move the current point at a soft restart, although it's not explicitly written in the documentation. 
Saving that evaluation might make sense because at a restart they increase the trust region radius to escape the current region. In that case the values in the trust region are not that interesting. Especially since one of the criteria causing a restart is that there have been only small function improvements lately.  

In any case, the default is True so unless a user explicitly changes this, moving is the behavior they will get.